packages/wonder-blocks-data/src/hooks/use-shared-cache.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import * as React from "react";
+import {KindError, Errors} from "@khanacademy/wonder-stuff-core";
+import {InMemoryCache} from "../util/in-memory-cache.js";
+import type {ValidData} from "../util/types.js";
+
+/**
+ * A function for inserting a value into the cache or clearing it.
+ */
+type CacheValueFn<TValue: ValidData> = (value: ?TValue) => void;
+
+/**
+ * This is the cache.
+ * It's incredibly complex.
+ * Very in-memory. So cache. Such complex. Wow.
+ */
+const cache = new InMemoryCache();
+
+/**
+ * Clear the in-memory cache or a single scope within it.
+ */
+export const clearSharedCache = (scope: string = "") => {
+    // If we have a valid scope (empty string is falsy), then clear that scope.
+    if (scope && typeof scope === "string") {
+        cache.purgeScope(scope);
+    } else {
+        // Just reset the object. This should be sufficient.
+        cache.purgeAll();
+    }
+};
+
+/**
+ * Hook to retrieve data from and store data in an in-memory cache.
+ *
+ * @returns {[?ReadOnlyCacheValue, CacheValueFn]}
+ * Returns an array containing the current cache entry (or undefined), a
+ * function to set the cache entry (passing null or undefined to this function
+ * will delete the entry).
+ *
+ * To clear a single scope within the cache or the entire cache,
+ * the `clearScopedCache` export is available.
```

Comment:
Nice.  It seems like there aren't many cases where you'd want to do that so not including it in the hook's API make sense.